terraform/provider/pkg/provider/provider.go

```diff
@@ -31,13 +31,7 @@ type PrivateKeyTFTokenProvider struct {
 	issuer, audience, scope string
 }
 
-func MakeKMSKeyTFProvider(ctx context.Context, provConfig *Config) (client.TokenProvider, error) {
-	cfg, err := config.LoadDefaultConfig(ctx)
-	if err != nil {
-		return nil, errors.Wrap(err, "unable to load SDK config")
-	}
-	stsClient := sts.NewFromConfig(cfg)
-	appCreds := stscreds.NewAssumeRoleProvider(stsClient, *provConfig.AssumeRoleARN)
+func MakeKMSKeyTFProvider(ctx context.Context, provConfig *Config, appCreds *stscreds.AssumeRoleProvider) (client.TokenProvider, error) {
```

Comment:
did this need to change?

Reply:
I changed it so the same `stscreds.AssumeRoleProvider` object could be used here and in the `TfAwsCredentialsProvider`. Is there a reason they should use different `stscreds.AssumeRoleProvider` objects?

Reply:
oo I see. I think Github was hiding a few other lines that were deleted and I didn't notice what was going on